### ISSUE TITLE
.iso extraction with parallel causes deadlock

### ISSUE THREAD

**Issue by svenclaesson** [microsoft/RecursiveExtractor#116]
Extracting the following file sometimes throws an error and sometimes deadlocks.
https://mirrors.slackware.com/slackware/slackware-iso/slackware64-14.1-iso/slackware64-14.1-install-dvd.iso

```
        [DataTestMethod]
        [DataRow("slackware64-14.1-install-dvd.iso", 8503)]
        public void ExtractIso(string fileName, int expectedNumFiles)
        {
            var extractor = new Extractor();
            var path = Path.Combine(Directory.GetCurrentDirectory(), "TestData", "TestDataArchives", fileName);
            var results = extractor.Extract(path,
                new ExtractorOptions()
                {
                    Parallel = true, BatchSize = 5, RawExtensions = new List<string>() { ".txz" }
                });
            Assert.AreEqual(expectedNumFiles, results.Count());
        }
```

With Parallel = false this completes in less then 20 seconds.
.txz files are not related, just excluded because it just takes extra time.

**Comment by gfs** [comment]
Will address soon.

**Comment by gfs** [comment]
Need one more change. Should be fixed with #118.
